src/Core/Commander/Providers/GPX_Provider.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Generated On: 2016-07-07
+ * Class: GPX_Provider
+ * Description: Parse GPX file to get [lat, lon, alt]
+ */
+
+define('Core/Commander/Providers/GPX_Provider', [
+        'Core/Commander/Providers/Provider',
+        'Core/Commander/Providers/IoDriverXML',
+        'THREE',
+        'Core/Geographic/CoordCarto'
+    ],
+    function(
+        Provider,
+        IoDriverXML,
+        THREE,
+        CoordCarto
+    ) {
+
+        function GPX_Provider(ellipsoid) {
+            //Constructor
+            this.ellipsoid = ellipsoid;
+            this.ioDriverGPX = new IoDriverXML();
+        }
+
+        GPX_Provider.prototype = Object.create(Provider.prototype);
+
+        GPX_Provider.prototype.constructor = GPX_Provider;
+
+        GPX_Provider.prototype.parseGPX = function(urlFile) {
+
+            return this.ioDriverGPX.read(urlFile).then(function(result) {
+
+                if (result === undefined)
+                    return undefined;
+
+                // ------------------------------------
+                // Getting the waypoint points
+                // ------------------------------------
+                var wpt = result.getElementsByTagName("wpt");
+                var geometry_p = new THREE.Geometry();
+
+                for (var i = 0; i < wpt.length; i++)
+                    geometry_p.vertices.push(this.ellipsoid.cartographicToCartesian(new CoordCarto().setFromDegreeGeo(Number(wpt[i].attributes.lon.nodeValue),Number(wpt[i].attributes.lat.nodeValue),Number(wpt[i].getElementsByTagName("ele")[0].childNodes[0].nodeValue))));
+
+                var material_p = new THREE.PointsMaterial( {color: 0x00ff00, size: 50, sizeAttenuation : true} );
```

Comment:
You probabty don't need to create 1 PointMaterial per point. 
Suggestion: move Materials creation before the ` for (var i = 0; i < wpt.length; i++)` loop